File: 807-custom-sort-string/custom-sort-string.py
class Solution:
    def customSortString(self, order: str, s: str) -> str:
        sa = ""
        remaining_chars = ""
        char_freq = {}
        for char in s:
            char_freq[char] = char_freq.get(char, 0) + 1
        for char in order:
            if char in char_freq:
                sa += char * char_freq[char]
                del char_freq[char]
        for char in char_freq:
            remaining_chars += char * char_freq[char]

        return sa + remaining_chars

File: 807-custom-sort-string/custom-sort-string.cpp
class Solution {
public:
    string customSortString(string order, string s) {
        ios_base::sync_with_stdio(false);
        cin.tie(NULL);
        cout.tie(NULL);
        string sa = "";
        string remaining_chars = "";
        unordered_map<char, int> char_freq;
        for (char c : s) {
            char_freq[c]++;
        }
        for (char c : order) {
            if (char_freq.find(c) != char_freq.end()) {
                sa.append(char_freq[c], c);
                char_freq.erase(c);
            }
        }
        for (auto& entry : char_freq) {
            remaining_chars.append(entry.second, entry.first);
        }
        return sa + remaining_chars;
    }
};